flink-formats/flink-csv/src/main/java/org/apache/flink/formats/csv/CsvRowDataSerializationSchema.java

```diff
@@ -0,0 +1,377 @@
+			}
+			this.csvSchema = this.csvSchema.rebuild().setLineSeparator(delimiter).build();
+			return this;
+		}
+
+		public Builder setArrayElementDelimiter(String delimiter) {
+			Preconditions.checkNotNull(delimiter, "Delimiter must not be null.");
+			this.csvSchema = this.csvSchema.rebuild().setArrayElementSeparator(delimiter).build();
+			return this;
+		}
+
+		public Builder disableQuoteCharacter() {
+			this.csvSchema = this.csvSchema.rebuild().disableQuoteChar().build();
+			return this;
+		}
+
+		public Builder setQuoteCharacter(char c) {
+			this.csvSchema = this.csvSchema.rebuild().setQuoteChar(c).build();
+			return this;
+		}
+
+		public Builder setEscapeCharacter(char c) {
+			this.csvSchema = this.csvSchema.rebuild().setEscapeChar(c).build();
+			return this;
+		}
+
+		public Builder setNullLiteral(String s) {
+			this.csvSchema = this.csvSchema.rebuild().setNullValue(s).build();
+			return this;
+		}
+
+		public CsvRowDataSerializationSchema build() {
+			return new CsvRowDataSerializationSchema(
+				rowType,
+				csvSchema);
+		}
+	}
+
+	@Override
+	public byte[] serialize(RowData row) {
+		if (root == null) {
+			root = csvMapper.createObjectNode();
+		}
+		try {
+			runtimeConverter.convert(csvMapper, root, row);
+			return objectWriter.writeValueAsBytes(root);
+		} catch (Throwable t) {
+			throw new RuntimeException("Could not serialize row '" + row + "'.", t);
+		}
+	}
+
+	@Override
+	public boolean equals(Object o) {
+		if (o == null || o.getClass() != this.getClass()) {
+			return false;
+		}
+		if (this == o) {
+			return true;
+		}
+		final CsvRowDataSerializationSchema that = (CsvRowDataSerializationSchema) o;
+		final CsvSchema otherSchema = that.csvSchema;
+
+		return rowType.equals(that.rowType) &&
+			csvSchema.getColumnSeparator() == otherSchema.getColumnSeparator() &&
+			Arrays.equals(csvSchema.getLineSeparator(), otherSchema.getLineSeparator()) &&
+			csvSchema.getArrayElementSeparator().equals(otherSchema.getArrayElementSeparator()) &&
+			csvSchema.getQuoteChar() == otherSchema.getQuoteChar() &&
+			csvSchema.getEscapeChar() == otherSchema.getEscapeChar() &&
+			Arrays.equals(csvSchema.getNullValue(), otherSchema.getNullValue());
+	}
+
+	@Override
+	public int hashCode() {
+		return Objects.hash(
+			rowType,
+			csvSchema.getColumnSeparator(),
+			csvSchema.getLineSeparator(),
+			csvSchema.getArrayElementSeparator(),
+			csvSchema.getQuoteChar(),
+			csvSchema.getEscapeChar(),
+			csvSchema.getNullValue());
+	}
+
+	// --------------------------------------------------------------------------------
+	// Runtime Converters
+	// --------------------------------------------------------------------------------
+
+	/**
+	 * Runtime converter that converts objects of Flink Table & SQL internal data structures
+	 * to corresponding {@link JsonNode}s.
+	 */
+	private interface SerializationRuntimeConverter extends Serializable {
```

Comment:
I think you should use something like `ParquetRowDataWriter.FieldWriter`.

Reply:
Do you mean the naming? I just want to follow the current naming style in csv and json formats.

Reply:
Or do you mean accessing fields of RowData using getters instead of `RowData.get` utiltiy? 
I thought about this, but due to we don't have `TypedGetters` interface for `RowData` and `ArrayData`, we have to implement two writers for each type. You can have a look at `org.apache.flink.table.runtime.arrow.writers.IntWriter`. This will be much more complex and we should come up with an idea how to reduce duplicate code. So I think that can be a future work.

Reply:
I don't think we should use `RowData.get(RowData row, int pos, LogicalType fieldType)` if we can have a better choice.
It is not only for the performance of csv, but also an important example for users.
A strong purpose of `RowData` is to recommend users to use specific `get` instead of inefficient and generic `get`.
We need to have a better way before the 1.11 release.